zio-cli/jvm/src/test/scala/zio/cli/FileBasedArgs.scala

```diff
@@ -0,0 +1,197 @@
+        homeDir <- ZIO.succeed(Paths.get(java.lang.System.getProperty("user.home")))
+        command <- ZIO.succeed("wc")
+
+        _ <- createLineCountTestFile(cwd, "sample_file")
+        _ <- createSampleConfigFiles(cwd, homeDir, command, content_home = "-w", content_cwd = "-l")
+
+        _      <- cliApp.run(args = List("sample_file")).either
+        output <- TestConsole.output
+
+        correctOutput <- ZIO.succeed("        3         4        20  sample_file\n")
+
+        _ <- cleanupLineCountTestFile(cwd, "sample_file")
+        _ <- cleanUpSampleConfigFiles(cwd: Path, homeDir: Path, command)
+      } yield assert(output.last)(equalTo(correctOutput))
+    },
+    test("should load options from files and merge them appropriatly") {
+      for {
+        // Create Sample config files
+        cwd     <- ZIO.succeed(Paths.get(java.lang.System.getProperty("user.dir")))
+        homeDir <- ZIO.succeed(Paths.get(java.lang.System.getProperty("user.home")))
+        command <- ZIO.succeed("testApp")
+        _       <- createSampleConfigFiles(cwd, homeDir, command)
+
+        // Check if the func checkAndGetOptionsFilePaths can
+        configArgs <- configFileOps.loadOptionsFromConfigFiles(command)
+
+        _ <- cleanUpSampleConfigFiles(cwd: Path, homeDir: Path, command)
+
+      } yield assert(configArgs)(hasSameElements(List("home=true", "dir=true", "home=false")))
+    },
+    test("should return directory ~/home and ./ which have .testApp config file for loading the args") {
+      for {
+        // Create Sample config files
+        cwd     <- ZIO.succeed(Paths.get(java.lang.System.getProperty("user.dir")))
+        homeDir <- ZIO.succeed(Paths.get(java.lang.System.getProperty("user.home")))
+        command <- ZIO.succeed("testApp1")
+        _       <- createSampleConfigFiles(cwd, homeDir, command)
+
+        // Check if the func checkAndGetOptionsFilePaths can
+        paths <- configFileOps.findPathsOfCliConfigFiles(command)
+
+        _ <- cleanUpSampleConfigFiles(cwd: Path, homeDir: Path, command)
+
+      } yield assert(paths)(hasSameElements(List(homeDir.toString(), cwd.toString())))
+    }
+  )
+
+  def createLineCountTestFile(
+    cwd: Path,
+    file_name: String = "sample_file",
+    content: String = "asdf\nqweqwer\njdsafn"
+  ): IO[IOException, Unit] =
+    ZIO.attempt {
+      Files.write(Paths.get(cwd.toString(), s"$file_name"), java.util.Arrays.asList(content));
+      ()
+    }.refineToOrDie[IOException]
+
+  def cleanupLineCountTestFile(cwd: Path, file_name: String = "sample_file"): IO[IOException, Unit] =
+    ZIO.attempt {
+      Files.delete(Paths.get(cwd.toString(), s"$file_name"));
+      ()
+    }.refineToOrDie[IOException]
+
+  def createSampleConfigFiles(
+    cwd: Path,
+    homeDir: Path,
+    command: String = "testApp",
+    content_home: String = "home=true",
+    content_cwd: String = "dir=true\nhome=false"
+  ): IO[IOException, Unit] =
```

Comment:
same thing as above, writing to file should return zio `Scope`, and clean itself up, doing so for a single file, and be called twice